apps/shopify/src/Pagination.js

```diff
@@ -25,8 +25,13 @@ class Pagination {
     this.shopifyClient = await makeShopifyClient(this.sdk);
   }
 
-  async fetchNext(search) {
+  async fetchNext(search, recursing = false) {
```

Comment:
I am not really a fan of arbitrary flags, but that might be me. What about interrupting recursion _before_ entering the new call? So just before the `return` of this method (L63 in the new version, L58 of the old one)

Reply:
Yeah, that seems like a good idea :)

Reply:
I'm not sure. I think that makes it more likely that we'll be returning data from a dead chain - and I think would make this change a lot more complicated.